src/util/websocket/Common.h

```diff
@@ -0,0 +1,98 @@
+//  Copyright 2023, University of Freiburg,
+//  Chair of Algorithms and Data Structures.
+//  Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#pragma once
+
+#include <cstdint>
+#include <random>
+#include <type_traits>
+
+#include "util/Exception.h"
+
+// Provides types required by all the other *.cpp files in this directory
+// and a select few other places
+namespace ad_utility::websocket::common {
+
+class QueryId {
+  std::string id_;
+  explicit QueryId(std::string id) : id_{std::move(id)} {}
+
+ public:
+  static QueryId idFromString(std::string id) { return QueryId{std::move(id)}; }
```

Comment:
This needs to be public, so `WebSocketManager` can create an instance from the URL the observing client used. This would conceptually a non-owning reference because it lives independently of the actual lifetime of the query